### PR TITLE
add task metrics for priority runner

### PR DESCRIPTION
**Pull request by glorv**
The different between multilevel and priority is the queuing strategy and the runner is almost the same, the only difference is that multilevel runner adjusts level 0 chance dynamically. So this PR extract the common part of the runner into a separate struct.

### CODE REVIEW

**Comment by glorv** [comment]
@BusyJay @Connor1996  PTAL

**Review by Connor1996 — DISMISSED** [review]
LGTM

File: src/queue.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ impl<T: TaskCell + Send> TaskInjector<T> {
         match self.0 {
             InjectorInner::SingleLevel(_) => Extras::single_level(),
             InjectorInner::Multilevel(_) => Extras::multilevel_default(),
-            InjectorInner::Priority(_) => Extras::single_level(),
+            InjectorInner::Priority(_) => Extras::multilevel_default(),
```
Comment (BusyJay) [inline src/queue.rs]: Better merge the same branch.

File: src/queue/multilevel.rs [hunk 2]
```diff
@@ -303,6 +293,142 @@ where
...
+
+    fn start(&mut self, local: &mut Local<T>) {
+        self.inner.start(local)
+    }
+
+    fn handle(&mut self, local: &mut Local<T>, task_cell: T) -> bool {
+        let res = self.inner.handle(local, task_cell);
+        let local_total = self.inner.local_total_elapsed_us.get();
+        if local_total > FLUSH_LOCAL_THRESHOLD_US {
             self.manager.maybe_adjust_chance();
```
Comment (BusyJay) [inline src/queue/multilevel.rs:450]: This will be unreachable because `local_total_elapsed_us` is flushed.

Comment (glorv) [reply src/queue/multilevel.rs:450]: fixed, PTAL again

**Review by BusyJay — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**